Automattic/wp-calypso
# Media: Populate advanced editing with parsed media

Related: #307, #2085

This pull request seeks to further progress advanced image editing by parsing the selected media item and adding the necessary behavior to update markup in the post editor after the user has chosen to save their changes.

![WIP](https://cloud.githubusercontent.com/assets/1779930/12399666/65be45c6-bde9-11e5-8de7-e76f73e7ed04.png)

*This is very much a work-in-progress.* As of now, the only expected behaviors are:
- A user can edit a selected image from the inline toolbar
- A modal appears, displaying the media's ID and a save button
- Upon clicking Save, the image is replaced with the text "Updated"

Note that this feature is feature-flagged, and is only available in a development environment. You can/should verify that these options are unavailable in all other environments.

__Implementation notes:__

The advanced media modal is no longer rendered by the `<PostEditor />` component, but instead by the `media/advanced` TinyMCE plugin. This is done so that the expected `insertMedia` prop handler can be passed to the modal which, when called, replaces the current selection with the passed markup.

__Testing instructions:__

1. Navigate to the [Calypso post editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Insert an image into the post contents
4. Select the image by clicking it
5. Click Edit in the inline toolbar
6. Note that a dialog is shown, containing the image's ID and a Save button
7. Click Save
8. Note that the modal is dismissed and that the image has been replaced in post contents with "Updated"
